ebean-core/src/main/java/io/ebeaninternal/server/transaction/JdbcTransaction.java

```diff
@@ -484,34 +484,28 @@ public final boolean isSaveAssocManyIntersection(String intersectionTable, Strin
     return existingBean.equals(beanName);
   }
 
-  /**
-   * Return the depth of the current persist request plus the diff. This has the
-   * effect of changing the current depth and returning the new value. Pass
-   * diff=0 to return the current depth.
-   * <p>
-   * The depth of 0 is for the initial persist request. It is modified as the
-   * cascading of the save or delete traverses to the the associated Ones (-1)
-   * and associated Manys (+1).
-   * </p>
-   * <p>
-   * The depth is used to help the ordering of batched statements.
-   * </p>
-   *
-   * @param diff the amount to add or subtract from the depth.
-   */
   @Override
   public final void depth(int diff) {
     depth += diff;
   }
 
-  /**
-   * Return the current depth.
-   */
   @Override
   public final int depth() {
     return depth;
   }
 
+  @Override
+  public final void depthDecrement() {
```

Comment:
this effectively means it won't take the depth negative.  So relative to other independent persist requests also at depth zero will execute in the order that they are persisted.  

Fixes testMultiSave4() and testMultiSave4_usingRelation4()